pkg/lib/sidecars/pods/get.go

```diff
@@ -51,32 +51,32 @@ func getAllRunningPods(ctx context.Context, c client.Client) (*v1.PodList, error
 	return podList, nil
 }
 
-func GetPodsToRestart(ctx context.Context, c client.Client, expectedImage SidecarImage, expectedResources v1.ResourceRequirements, predicates []filter.SidecarProxyPredicate, logger *logr.Logger) (outputPodsList v1.PodList, err error) {
+func GetPodsToRestart(ctx context.Context, c client.Client, expectedImage SidecarImage, expectedResources v1.ResourceRequirements, predicates []filter.SidecarProxyPredicate, logger *logr.Logger) (outputPodsList *v1.PodList, err error) {
 	podList, err := getAllRunningPods(ctx, c)
 	if err != nil {
-		return outputPodsList, err
+		return nil, err
 	}
 
-	podList.DeepCopyInto(&outputPodsList)
-	outputPodsList.Items = []v1.Pod{}
-
 	//Add predicate for image version and resources configuration
 	predicates = append(predicates, NewRestartProxyPredicate(expectedImage, expectedResources))
 
 	for _, predicate := range predicates {
 		evaluator, err := predicate.NewProxyRestartEvaluator(ctx)
 		if err != nil {
-			return v1.PodList{}, err
+			return &v1.PodList{}, err
 		}
 
+		outputPodsList = &v1.PodList{}
 		for _, pod := range podList.Items {
 			if evaluator.RequiresProxyRestart(pod) {
-				outputPodsList.Items = append(outputPodsList.Items, *pod.DeepCopy())
+				outputPodsList.Items = append(outputPodsList.Items, pod)
```

Comment:
Maybe a micro-optimization since the pod objects are not that big, but I think we will create multiple backing arrays, since the capacity of the slices will increase multiple times. That will lead to multiple arrays being kept in memory until the outputPodList isn't referenced any longer and can be garbage collected.
We could initialize the `outputPodsList.Items` with a capacity derived from podList.Items length.